.github/workflows/ci-tests.yml

```
@@ -0,0 +1,182 @@
+name: Continous integration tests
+
+on:
+  push:
+  pull_request:
+
+env:
+  singularity_version: 3.6.4
+
+jobs:
+
+  tox:
+    name: CI tests via Tox
+
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        py-ver-major: [3]
+        py-ver-minor: [6, 7, 8, 9]
+        step: [lint, unit, bandit, mypy]
+        exclude:
+          - py-ver-major: 3
+            py-ver-minor: 9
+            step: mypy
+
+    env:
+      py-semver: ${{ format('{0}.{1}', matrix.py-ver-major, matrix.py-ver-minor) }}
+      TOXENV: ${{ format('py{0}{1}-{2}', matrix.py-ver-major, matrix.py-ver-minor, matrix.step) }}
+
+    steps:
+      - uses: actions/checkout@v2
```

```suggestion
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

As per https://github.com/python/importlib_metadata/pull/252/files

---

Good spot. I was very confused as to why this failed. Never forget to RTFM.

---

Actually, I think you mean to do this on line 92?

---

> Good spot. I was very confused as to why this failed. Never forget to RTFM.

I just did a web search for the error message :-P

---

It's not resolved, but by replaying the git commands from CI I can reproduce the error locally. The diff command is incorrect cos `main` isn't a local branch. Is there any problem setting `--compare-branch=origin/main` in the Makefile for `diff_pydocstyle_report`?

---

Or I guess I could just add `git branch main origin/main` before running?